tree/treeplayer/inc/ROOT/TInterface.hxx

```diff
@@ -1083,11 +1083,11 @@ protected:
 
          ForeachSlot(fillTree, {bnames[S]...});
 
-         for (auto &&tree : trees) tree->Write();
-         for (auto &&file : files) file->Write();
-
-         trees.clear();
-         files.clear();
+         for (unsigned int slot = 0; slot < nSlots; ++slot) {
+            files[slot]->cd();
+            trees[slot]->Write();
```

Comment:
The cd+tree->Write are redundant.  The file Write should be enough (it goes though the list of things owned by the TFile and write them).